### PR TITLE
test: add tests to LSP23

### PR DESCRIPTION
**Pull request by skimaharvey**
# What does this PR introduce? 

- [x] Add tests for proxy deployments

### CODE REVIEW

File: contracts/Mocks/Reentrancy/ReceiveContract.sol [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+contract ReceiveContract {
+    event Received(address, uint);
```
Comment (CJ42) [inline contracts/Mocks/Reentrancy/ReceiveContract.sol:5]: Suggesting to be explicit with the type.

```
suggestion
    event Received(address, uint256);
```

File: tests/LSP23LinkedContractsDeployment/helpers.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
-import { ethers } from 'ethers';
+import { ethers } from 'hardhat';
```
Comment (CJ42) [inline tests/LSP23LinkedContractsDeployment/helpers.ts:1]: Is there a difference to import ethers from "ether" vs "hardhat"?

File: tests/LSP23LinkedContractsDeployment/helpers.ts [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+import { BytesLike } from 'ethers';
```
Comment (CJ42) [inline tests/LSP23LinkedContractsDeployment/helpers.ts:2]: Nice 👌🏻

File: tests/LSP23LinkedContractsDeployment/helpers.ts [hunk 4]
```diff
@@ -48,3 +49,36 @@ export async function calculateProxiesAddresses(
 
   return [expectedPrimaryContractAddress, expectedSecondaryContractAddress];
 }
+
+export function createDataKey(prefix, address) {
```
Comment (CJ42) [inline tests/LSP23LinkedContractsDeployment/helpers.ts:53]: I am not sure to understand. Where are these 2 functions used?

Comment (skimaharvey) [reply tests/LSP23LinkedContractsDeployment/helpers.ts:53]: they are used in my tests

File: contracts/Mocks/Reentrancy/ReceiveContract.sol (hunk 1 above)
Comment (skimaharvey) [reply contracts/Mocks/Reentrancy/ReceiveContract.sol]: cool. Deleted this contract

**Review by b00ste — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**